root/etc/NetworkManager/dispatcher.d/15-get-zip-server

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+# Script to extract the NEXT_SERVER value from the incoming DHCP lease
+# and write it to a place where systemd/profile.d can source it
+
+# This is only run when the interface is going "up" so we don't interfere
+# with the next-server settings on "down"
+
+zipserver=""
+
+# create/replace file using a tempfile.
+# This allows us to trigger on PathExists inside systemd.path units.
+function tmpwriter {
+  tmpfile="$(TMPDIR="${1%/*}" mktemp)"
+  cat > "${tmpfile}"
+  mv "${tmpfile}" "${1}"
+}
+
+if [ $# -ne 2 ]
+then
+  echo "usage: $0 <iface> <action>" >&2
+  echo '  Only action "up" is used by this script.' >&2
+  exit 1
+fi
+
+if [ "${2}" != "up" ]
+then
+  exit
+fi
+
+iface="${1}"
+
+read hwaddr < "/sys/class/net/${iface}/address"
+
+read -a cmdline < /proc/cmdline
+for param in "${cmdline[@]}"
+do
+  value="${param#*=}"
+  case "${param%%=*}" in
+    "fdi.zipserver")
+      # retrieve zipserver from the kernel cmdline
+      zipserver="${value#*=}"
+    ;;
+    "BOOTIF")
+      # normalization of MAC addresses (lower case, separated by :)
+      # strip out leading arp type (01-)
+      bootif="$(tr A-F- a-f: <<< "${value#*-}")"
+    ;;
+  esac
+done
```

Comment:
Nitpick: Can you make this a generic function putting it in `/usr/share/fdi` perhaps? All `fdi.something` values setting as shell variables.  This could be reused in nm-prepare.